pages/thank-you.jsx

```diff
@@ -1,27 +1,40 @@
-// this will ignore that Bad Invocation error which is a bug in JSXHint
-/* jshint -W067 */
-
 import React from 'react';
 import MozillaFooter from '../components/footer-mozilla.jsx';
 import Signup from '../components/signup.jsx';
 import Social from '../components/social.jsx';
 import ThankYouHeader from '../components/thank-you-header.jsx';
-import { IntlMixin } from 'react-intl';
 import analytics from '../assets/js/analytics.js';
 import form from '../scripts/form.js';
+import qs from 'qs';
 
 var ThankYou = React.createClass({
-  mixins: [IntlMixin],
+  contextTypes: {
+    location: React.PropTypes.object,
+    router: React.PropTypes.object,
+    intl: React.PropTypes.object
+  },
+  propTypes: {
+    country: React.PropTypes.string.isRequired,
+    test: React.PropTypes.string,
+    location: React.PropTypes.object
+  },
+  getInitialState: function() {
+    let query = qs.parse(this.props.location.search.replace("?", ""));
+    return {
+      email: query.email
+    };
+  },
   componentDidMount: function() {
-    form.updateField("email", this.props.email || "");
+    form.updateField("email", this.state.email || "");
     analytics();
   },
   render: function() {
     var className = "row thank-you-page";
-    var locale = this.props.locales[0];
-    var signUpOrSocial = (<Social language={locale}/>);
-    if (this.props.params && /^(en|de)(\b|$)/.test(locale)) {
-      signUpOrSocial = (<Signup country={this.props.country} email={this.props.email} locales={this.props.locales} />);
+    var locale = this.context.intl.locale;
+    var signUpOrSocial = (<Social />);
+
+    if (/^(en|de)(\b|$)/.test(locale)) {
```

Comment:
I followed it from when we first added it, and `this.props.params` has always been there, and never looks like it's doing anything, so I'm not feeling a lot of confidence in removing it until I know what it was doing.